sefaria/model/text.py

```diff
@@ -4442,6 +4515,9 @@ def part_projection(self):
         # todo: reimplement w/ aggregation pipeline (see above)
         # todo: special case string 0?
 
+        if self.index_node.is_virtual:
```

Comment:
where are we handling virtual nodes?

Reply:
in the text property of TextRange. i'm adding a test for that.